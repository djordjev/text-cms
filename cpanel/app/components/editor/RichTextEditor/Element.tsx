import classnames from 'classnames';
import React from 'react';
import { RenderElementProps } from 'slate-react';

export interface ElementProps extends RenderElementProps {}

const Element: React.FC<ElementProps> = (props) => {
  const { attributes, children, element } = props;

  const { align } = element;

  // Styles
  const classes = classnames({
    'u-text-left': !align || align === 'left',
    'u-text-center': align === 'center',
    'u-text-right': align === 'right'
  });

  // Markup
  const renderHeading = (level: number) => {
    // prettier-ignore
    if (level === 1) return <h1 {...attributes} style={{ fontSize: '64px'}}>{children}</h1>;
    // prettier-ignore
    if (level === 2) return <h2 {...attributes} style={{ fontSize: '32px'}}>{children}</h2>;
    // prettier-ignore
    if (level === 3) return <h3 {...attributes} style={{ fontSize: '24px'}}>{children}</h3>;
    // prettier-ignore
    if (level === 4) return <h4 {...attributes} style={{ fontSize: '20px'}}>{children}</h4>;
    // prettier-ignore
    if (level === 5) return <h5 {...attributes} style={{ fontSize: '16px'}}>{children}</h5>;
    // prettier-ignore
    if (level === 6) return <h6 {...attributes} style={{ fontSize: '16px'}}>{children}</h6>;
  };

  if (element.type === 'heading') {
    return <div className={classes}>{renderHeading(element.level)}</div>;
  }

  return (
    <p className={classes} {...attributes}>
      {children}
    </p>
  );
};

export { Element };
