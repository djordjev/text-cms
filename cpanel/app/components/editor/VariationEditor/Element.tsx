import React from 'react';
import { RenderElementProps } from 'slate-react';

const Element: React.FC<RenderElementProps> = (props) => {
  const { attributes, children, element } = props;

  // Markup
  const renderHeading = (level: number) => {
    // prettier-ignore
    if (level === 1) return <h1 {...attributes} style={{ fontSize: '64px'}}>{children}</h1>;
    // prettier-ignore
    if (level === 2) return <h2 {...attributes} style={{ fontSize: '32px'}}>{children}</h2>;
    // prettier-ignore
    if (level === 3) return <h3 {...attributes} style={{ fontSize: '24px'}}>{children}</h3>;
    // prettier-ignore
    if (level === 4) return <h4 {...attributes} style={{ fontSize: '20px'}}>{children}</h4>;
    // prettier-ignore
    if (level === 5) return <h5 {...attributes} style={{ fontSize: '16px'}}>{children}</h5>;
    // prettier-ignore
    if (level === 6) return <h6 {...attributes} style={{ fontSize: '16px'}}>{children}</h6>;
  };

  if (element.type === 'heading') {
    return renderHeading(element.level);
  }

  return <p {...attributes}>{children}</p>;
};

export { Element };
