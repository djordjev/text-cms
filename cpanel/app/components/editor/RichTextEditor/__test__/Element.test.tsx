import { createRemixStub } from '@remix-run/testing';
import { render } from '@testing-library/react';

import { Element, ElementProps } from '../Element';

test('Element', () => {
  const attributes: any = {};
  const children: unknown = [];

  const props: ElementProps = {
    attributes,
    children,
    element: {
      type: 'paragraph',
      children: [{ text: 'children-text' }]
    }
  };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <Element {...props} /> }
    ]);

    return <RemixStub />;
  };

  // Paragraph
  const { container, rerender } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot('paragraph');

  // Heading
  props.element = { align: 'right', children: [], level: 2, type: 'heading' };
  rerender(createComponent());
  expect(container.childNodes).toMatchSnapshot('heading');
});
