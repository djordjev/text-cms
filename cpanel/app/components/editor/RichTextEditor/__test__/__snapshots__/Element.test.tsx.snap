// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Element > heading 1`] = `
NodeList [
  <div
    class="u-text-right"
  >
    <h2
      style="font-size: 32px;"
    />
  </div>,
]
`;

exports[`Element > paragraph 1`] = `
NodeList [
  <p
    class="u-text-left"
  />,
]
`;
